lib/errors/index.js

```diff
@@ -226,21 +226,96 @@ exports.UnknownConstraintError = UnknownConstraintError;
+
+    this.origin         = null;
+    this.instance       = inst || null;
+    this.validatorKey   = validatorKey || null;
+    this.validatorName  = fnName || null;
+    this.validatorArgs  = fnArgs || [];
+
+    if (type) {
+      if (ValidationErrorItem.ORIGINS[ type ]) {
+        this.origin = type;
+      } else {
+        const lowercaseType = (type + '').toLowerCase().trim();
+        const realType  = ValidationErrorItem.TypeStringMap[ lowercaseType ];
+
+        if (realType && ValidationErrorItem.ORIGINS[ realType ]) {
+          this.origin = realType;
+          this.type = type;
+        }
+      }
+    }
+  }
+
+  /**
+   * return a lowercase, trimmed string "key" that identifies the validator.
+   *
+   * N.B. the string will be empty if the instance has neither a valid `validatorKey` property nor a valid `validatorName` property
+   *
+   * @param   {Boolean} useTypeAsNS   - optional, defaults to TRUE, controls whether the returned value is "namespace",
+   *                                    this parameter is ignored if the validator's `type` is not one of ValidationErrorItem.ORIGINS
+   * @param   {String}  NSSeparator   - a separator string for concatenating the namespace, must be not be empty,
+   *                                    defaults to "." (fullstop). only used and validated if useTypeAsNS is TRUE.
+   * @throws  {Error}                 - thrown if NSSeparator is found to be invalid.
+   * @return  {String}
+   */
+  getValidatorKey(useTypeAsNS/* = true */, NSSeparator/* = '.'*/) {
+    // parameter defaults: some CI instances give a syntax error on actual parameter default values
+    const
+      useTANS = typeof useTypeAsNS === 'undefined' ?  true : !!useTypeAsNS,
+      NSSep   = typeof NSSeparator === 'undefined' ? '.' : NSSeparator;
+
+    const
+      type  = this.origin,
+      key   = this.validatorKey || this.validatorName,
+      useNS = useTANS && type && ValidationErrorItem.ORIGINS[ type ];
+
+    if (useNS && (typeof NSSep !== 'string' || !NSSep.length)) {
+      throw new Error('Invalid namespace separator given, must be a non-empty string');
+    }
+
+    if (!(typeof key === 'string' && key.length)) {
+      return '';
+    }
+
+    return (useNS ? [type, key].join(NSSep) : key).trim().toLowerCase();
   }
 }
-exports.ValidationErrorItem = ValidationErrorItem;
+exports.ValidationErrorItem       = ValidationErrorItem;
+
+ValidationErrorItem.ORIGINS         = Object.freeze({
+  // fValidation errors that orignate from checks thats Sequelize does based on 'core' attribute definitions (e.g. "notNull")
+  CORE                            : 'CORE',
+  // Validation errors that originate from the storage engine
+  DB                              : 'DB',
+  // Validation errors that originate from validator functions (both built-in and custom) defined for a given attribute
+  FUNCTION                        : 'FUNCTION'
+});
+
+ValidationErrorItem.TypeStringMap = {
```

Comment:
removing `type` in the 4/5 places it is being given the `ValidationErrorItem` constructor is slightly more work because you also have to change the constructor signature. 

the way it is now you simply replace the literal string value with a reference to something like `sequelizeError.ValidationErrorItem.ORIGINS.FUNCTION`, e.g. effectively remove the comment and everything before it on the changed lines: 

https://github.com/sequelize/sequelize/pull/7557/commits/9952b7c1284bf3924e397d3a77827ea42a57b14e

once you make that replacement, the map is no longer used (and `ValidationErrorItem.type` will no longer be set)